### PR TITLE
Health api copy editing

### PR DESCRIPTION
**Pull request by jbaiera**
Edits to human readable text returned from the health api. A number of edits have been added for readability and correctness.

### CODE REVIEW

**Review by andreidan — APPROVED** [review]
LGTM, thanks Jimmy

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java [hunk 1]
```diff
@@ -279,11 +281,11 @@ public HealthIndicatorResult calculate(boolean explain) {
                 tier -> tier,
                 tier -> new UserAction.Definition(
                     "increase_tier_capacity_for_allocations_" + tier,
-                    "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in their data tiers "
+                    "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in the desired data tier "
```
Comment (cristina-eleonora) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java:284]: If the user is using multiple tiers, this information might not be enough. Could we specify which tier we're talking about?

File: server/src/main/java/org/elasticsearch/health/HealthService.java [hunk 2]
```diff
@@ -35,11 +35,11 @@
 public class HealthService {
 
     // Visible for testing
-    static final String UNKNOWN_RESULT_SUMMARY_PREFLIGHT_FAILED = "Could not determine indicator state. Cluster state is not stable. Check "
-        + "details for critical issues keeping this indicator from running.";
+    static final String UNKNOWN_RESULT_SUMMARY_PREFLIGHT_FAILED = "Could not determine health status. Check details on critical issues "
```
Comment (cristina-eleonora) [inline server/src/main/java/org/elasticsearch/health/HealthService.java:38]: We should specify for which indicator we cannot determine the status. 

e.g.: "Could not determine shard allocation health status..."

File: server/src/main/java/org/elasticsearch/health/HealthService.java [hunk 3]
```diff
@@ -35,11 +35,11 @@
...
+        + "preventing the health status from reporting.";
     static final String UNKNOWN_RESULT_SUMMARY_NOT_RECOVERED =
-        "Could not determine indicator state. The current node handling the health request is not ready to assess the health of the "
-            + "cluster. Try again later or execute the health API against a different node.";
+        "Could not determine health status. This node is not ready to assess the cluster health. Try again later or run the health API "
```
Comment (cristina-eleonora) [inline server/src/main/java/org/elasticsearch/health/HealthService.java:41]: Same comment as above. We should specify for which indicator we cannot determine the status.

e.g.: "Could not determine shard allocation health status..."

Also, we should say which node is not ready to asses the cluster health. Perhaps this phrase can be further simplified. e.g.: "Node {node_ID} cannot assess the cluster health."

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java (hunk 1 above)
Comment (jbaiera) [reply server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java:293]: We discussed off issue and decided to move this to the next edit PR.

File: server/src/main/java/org/elasticsearch/health/HealthService.java (hunk 2 above)
Comment (jbaiera) [reply server/src/main/java/org/elasticsearch/health/HealthService.java:38]: We discussed off-issue and decided to move this to the next edit PR.

File: server/src/main/java/org/elasticsearch/health/HealthService.java (hunk 3 above)
Comment (jbaiera) [reply server/src/main/java/org/elasticsearch/health/HealthService.java:41]: We discussed off-issue and decided to move this to the next edit PR.

**Pull request merged**